### PR TITLE
Circleci runs pre-commit remove duplicate tests

### PR DESCRIPTION
**Pull request by cclauss**
* #2741

But also simplifies configuration by removing duplicate tests that are already being run by `pre-commit`.

### CODE REVIEW

**Review by timobrembeck — COMMENTED** [review]
Thanks a lot, yes this makes sense! :+1:

I already had a similar change in mind, also because it was cumbersome to keep the version definitions in the pre-commit config in sync with the rest of the project. Maybe we can even get rid of these dependencies in the pyproject.toml now? Since pre-commit installs them separately? We would just also have to remove all the dev tools that depend on them. But since pre-commit can also run single checks, I don't see an important reason to keep them.

I had concerns about the run time of the checks, since we loose the ability to run the checks in parallel, but according to CircleCI, it doesn't make a huge difference. Maybe due to the higher resource class, but I guess we can afford it.

Sorry about the new conflicts with the ruff tool, do you think you could resolve them?
In any case, I'm very grateful for all your contributions! :pray:

**Comment by cclauss** [comment]
The lack of a performance delta might be less about the large memory size and more about avoiding cloning the repo over and over again.

> Maybe we can even get rid of these dependencies in the pyproject.toml now? Since pre-commit installs them separately?

Yes. Yes.

**Comment by cclauss** [comment]
Given how long it takes to get pull requests reviewed and merged, I hesitate to add more complexity here.
@PeterNerlich

**Comment by charludo** [comment]
> Given how long it takes to get pull requests reviewed and merged, I hesitate to add more complexity here. Small moves. @PeterNerlich

Yes, sorry, that was more of a general remark regarding the idea of removing the devtools :)

**Comment by PeterNerlich** [comment]
> Given how long it takes to get pull requests reviewed and merged, I hesitate to add more complexity here. Small moves. @PeterNerlich

> Yes, sorry, that was more of a general remark regarding the idea of removing the devtools :)

Sure, let's keep this idea in mind for a future PR.

**Review by deen13 — CHANGES_REQUESTED** [review]
Thanks for the pull request! I really like the idea of running all pre-commits in circle ci. ☺️

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,38 +30,10 @@ jobs:
...
-          name: Enable virtual environment
-          command: echo "source .venv/bin/activate" >> $BASH_ENV
-      - run:
-          name: Check formatting of Django templates
-          command: djlint --check --lint integreat_cms
-  isort:
-    docker:
-      - image: cimg/python:3.11.7
-    resource_class: small
+    resource_class: large
```
[the remark this answers is not in the document]
Comment (cclauss) [reply .circleci/config.yml:36]: I believe that it was MyPy that required the extra memory.  Try moving that to a separate job and removing it from pre-commit and dropping back to a small instance.  Much is written about why running MyPy in pre-commit is bad.

**Pull request closed without merging**